### DISCUSSION TITLE
There is a way to test Avalonia.LinuxFramebuffer features in ubuntu ?

### DISCUSSION THREAD

**Discussion by zxslzx** [AvaloniaUI/Avalonia#7340] [Q&A]
I tested it with ubuntu 20.04 in vmware, There is no error, But it didn't display.
![image](https://user-images.githubusercontent.com/32009362/148870702-ebdb73b8-9c90-475f-9e86-7187dcc68391.png)

**Comment by kekekeks** [comment]
For testing with Ubuntu we were using:
1) Ubuntu 18.04 running bare metal on touch-capable laptops (Dell Precision 7510, Dell Precision 7530)
2) Ubuntu 18.04 running in KVM virtual machine with "QEMU USB Tablet" input

Your particular Hypervisor might not provide a compatible framebuffer.
Try running:
1) `fbset` - it should print framebuffer information
2) `cat /dev/urandom > /dev/fb0` - it should fill framebuffer with data

Also consider using DRM backend if your emulated hardware supports DRM/KMS/GBM/EGL or switching to another hypervisor.

**Reply by zxslzx** [reply]
1. fbset
![image](https://user-images.githubusercontent.com/32009362/148904489-8de49867-640f-4922-a911-98af46237c8d.png)
2. cat /dev/urandom > /dev/fb0
![image](https://user-images.githubusercontent.com/32009362/148904598-c0ec8195-638b-4c1f-bc24-f425b78c3214.png)

**Reply by kekekeks** [reply]
Your framebuffer looks compatible. Have you tried running ControlCatalog.NetCore with `--fbdev`?

**Reply by zxslzx** [reply]
I tried to run ControlCatalog.NetCore and it worked properly.
In addition,It seems impossible to input using the keyboard.
![image](https://user-images.githubusercontent.com/32009362/149043836-7471bfbf-9a93-4990-8816-d49a5d948da8.png)
![image](https://user-images.githubusercontent.com/32009362/149044141-0a418b99-9042-47b9-8f59-5078f3f6defb.png)

**Reply by kekekeks** [reply]
You are probably missing something in your configuration (usually it's checking for ISingleViewApplicationLifetime in App.xaml.cs).

Regarding keyboard input: framebuffer platform is currently optimized for touch-capable devices without keyboard.
